services/job.h

```diff
@@ -29,9 +29,10 @@
 #include <sstream>
 
 typedef enum {
-    Arg_Local,  // Local-only args.
-    Arg_Remote, // Remote-only args.
-    Arg_Rest    // Args to use both locally and remotely.
+    Arg_Local = 0x1,               // Local-only args.
+    Arg_Remote = 0x2,              // Remote-only args.
+    Arg_RestAndDwarfFission = 0x4, // like Arg_Rest, but only if DwarfFission is enabled.
```

Comment:
Why is there a separate type for -gsplit-dwarf? As far as I can judge it can be normally included in Arg_Rest.

Reply:
It is required to keep the current behavior of adding -gsplit-dwarf only when compiling. The flag is dropped in all other cases. It looks like this was added intentionally, so I tried to keep it. I guess this is for clang, which complains when adding the flag without compiling.

Reply:
I see. But if the flag is to be used only for compiling, then it should work to keep it in Arg_Remote (which is a bit of a misnomer, Arg_Local technically means "for preprocessing" and Arg_Remote means "for compiling"). Does it work that way?